test/index.test.js

```diff
@@ -156,41 +142,99 @@ describe('test', function () {
     })
   })
 
-  describe('generic post', async () => {
-    it('calls reauthenticate when the access token is expired', async () => {
-      wrapper = rewire('../index.js')
-      wrapper.config({ key: credsKey, secret: credsSecret, base: credsBase })
-      const reauthenticate = wrapper.__get__('_reauthenticate')
-      const reauthenticateSpy = sinon.spy(reauthenticate)
-      wrapper.__set__('_reauthenticate', reauthenticateSpy)
-      const data = { books: ['the', 'books'] }
-      // authenticate
+  describe('reauthentication for expired access token', () => {
+    let handleAuthErrorSpy
+    const path = 'path'
+    beforeEach(() => {
+      const _handleAuthError = wrapper.__get__('_handleAuthError')
+      handleAuthErrorSpy = sinon.spy(_handleAuthError)
+      wrapper.__set__('_handleAuthError', handleAuthErrorSpy)
+      // mock successful auth call
       mockAxios.onPost(`${credsBase}token`, auth)
```

Comment:
I don't think this `onPost` is matching anything. We don't authenticate by posting a JSON body with username and password, so I don't think `mockAxios.onPost(`${credsBase}token`, auth)` is matching the token request. If you inspect the `mockAxios.history`, I think you'll see the client is not using the mocked access token.